### PR TITLE
I-ALiRT - Incorporate uksa schedule

### LINKED ISSUE #2683: I-ALiRT - Create ground station schedule
### Summary

_No response_

### Requirements

The more specific the better.
  - [ ]
  - [ ]

### Related tickets

_No response_

### PR DESCRIPTION
**Pull request by laspsandoval**
# Change Summary

## Overview

UKSA would like to send us a schedule. Therefore, I've created a way to parse their schedule so that it will appear here when we finalized their partnership.
https://imap-mission.com/ialirt/ground-station

## File changes

**UKSA schedule integration:**

* Added the `parse_uksa_schedule_xlsx` function in `generate_coverage.py` to parse UKSA station contact windows from an Excel file, including logic to handle setup and teardown times based on notes in the sheet.
* Updated the `generate_coverage` function to accept a `uksa` parameter and incorporate UKSA contacts into the coverage and outage calculations.

**Testing improvements:**

* Introduced new tests in `test_generate_coverage.py` to verify UKSA schedule parsing and correct integration of UKSA coverage, including edge cases for setup and teardown adjustments.
* Added a fixture to provide the UKSA schedule path for testing.

### CODE REVIEW

File: imap_processing/ialirt/generate_coverage.py [hunk 1]
```diff
@@ -77,10 +145,11 @@ def create_schedule_mask(
     return schedule_mask
 
 
-def generate_coverage(
+def generate_coverage(  # noqa: PLR0912
     start_time: str,
     outages: dict | None = None,
     dsn: dict | None = None,
+    uksa: list | None = None,
```
Comment (tech3371) [inline imap_processing/ialirt/generate_coverage.py:152]: should this be generalized for future new partner's schedule? Eg. similar to your other input, make that input be dictionary of `{partner_name: list, partner_name: list}`

**Review by tech3371 — APPROVED** [review]
Code was easy to follow. Nice work, Laura!
